.pre-commit-config.yaml

```diff
@@ -0,0 +1,12 @@
+# See https://pre-commit.com for more information
+# See https://pre-commit.com/hooks.html for more hooks
+repos:
```

Comment:
@philschatz maybe we should add the faster checks like `tsc` and `tslint` as commit hooks?

Reply:
I really like the use of pre-commit hooks. I think they are a great way to run the checks as you commit instead of waiting for some other time. Even though I've had linters run based on the test command I always seem to break CI for some dumb linting error. Now that I use pre-commit hooks it doesn't happen anymore.

Reply:
i would definitely not want a pre-commit hook to run tests because i don't want to wait for them every time i commit. the linter is pretty fast though. i like to have control over when my code gets re-formatted and would not love to have it happen automatically on commit

Reply:
yea, i don't run tests.. i don't like that. only linting checks on the code i'm committing. We can just use this for the python stuff. I can move it to the python Makefile.